vcd/resource_vcd_vapp_vm.go

```diff
@@ -926,16 +1005,38 @@ func resourceVcdVAppVmUpdateExecute(d *schema.ResourceData, meta interface{}) er
 
 	}
 
-	if d.HasChanges("memory", "cpus", "cpu_cores", "power_on", "disk", "expose_hardware_virtualization",
-		"network", "boot_image", "hardware_version", "os_type", "description") {
+	memoryNeedsColdChange := false
+	cpusNeedsColdChange := false
+	networksNeedsColdChange := false
+	if executionType == "update" {
+		if !d.Get("memory_hot_add_enabled").(bool) && d.HasChange("memory") {
+			memoryNeedsColdChange = true
+		}
+		if !d.Get("cpu_hot_add_enabled").(bool) && d.HasChange("cpus") {
+			cpusNeedsColdChange = true
+		}
+		if isNetworkRemoved(d) && d.HasChange("network") {
+			networksNeedsColdChange = true
+		}
+	} else {
+		// if create happens allow to process it, keeping it as it was
+		networksNeedsColdChange = true
+	}
+
+	if d.HasChanges("cpu_cores", "power_on", "disk", "expose_hardware_virtualization", "boot_image",
```

Comment:
Please add a comment above to explain that the purpose of this if is to determine if parameters were changed which require VM reboot, and whatever else would be important to know.

One more thought: if powering off the VM is the only purpose here, what about moving this whole block to a separate function something like `undeployVMIfNeeded`.

Reply:
Firstly change is needed is to split create and update

Reply:
comment added